src/box/lua/schema.lua

```diff
@@ -330,22 +342,61 @@ local function feedback_save_event(event)
     end
 end
 
+-- Extract isolation names from C, create and return name->ID map.
+local function collect_txn_isolation()
+    local res = {}
+    for i = 0, builtin.txn_isolation_level_MAX - 1 do
+        res[ffi.string(builtin.txn_isolation_level_strs[i])] = i
+    end
+    return res;
+end
+
+-- Convert to numeric the value of tnx isolation, raise an error if failed.
+local function normalize_txn_isolation(txn_isolation)
+    if type(txn_isolation) == 'string' then
+        txn_isolation = box.txn_isolation[txn_isolation]
+        if not txn_isolation then
+            box.error(box.error.ILLEGAL_PARAMS,
+                      "txn_isolation must be one of box.txn_isolation" ..
+                      " (keys or values)")
+        end
+    elseif type(txn_isolation) ~= 'number' then
+        box.error(box.error.ILLEGAL_PARAMS,
+                  "txn_isolation must be a string or number")
+    end
+    return txn_isolation
+end
+
+box.txn_isolation = collect_txn_isolation()
+
+box.internal.normalize_txn_isolation = normalize_txn_isolation
+
 box.begin = function(options)
     local timeout
+    local txn_isolation
     if options then
         check_param(options, 'options', 'table')
         timeout = options.timeout
         if timeout and (type(timeout) ~= "number" or timeout <= 0) then
             box.error(box.error.ILLEGAL_PARAMS,
                       "timeout must be a number greater than 0")
         end
+        txn_isolation = options.txn_isolation
+        if txn_isolation ~= nil then
```

Comment:
Nit:
```suggestion
        if txn_isolation  then
```

Reply:
@CuriousGeorgiy, 
> Avoid implicit casting to boolean in if conditions like if x then or if not x then. Such expressions will likely result in troubles with box.NULL. Instead of those conditions, use if x ~= nil then and if x == nil then.